integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -2075,4 +2081,400 @@ describe("Flexible sync", function () {
+            callback,
+          );
+
+          const subscription = await realm.objects(Person).subscribe();
+
+          realm.write(() => {
+            realm.create(Person, {
+              _id: new BSON.ObjectId(),
+              name: "Heavy",
+              age: 36,
+              data: new ArrayBuffer(1_000_000),
+            });
+          });
+
+          await realm.syncSession?.uploadAllLocalChanges();
+          const subscription2 = await realm2.objects(Person).subscribe();
+          await realm2.syncSession?.downloadAllServerChanges();
+
+          expect(callback.args.find(([estimate]) => estimate < 1)).to.not.be.undefined;
+          expect(callback.withArgs(1.0).callCount).is.greaterThanOrEqual(2);
+
+          subscription.unsubscribe();
+          subscription2.unsubscribe();
+          realm2.close();
+        });
+
+        it("should not run after it has been removed", async function (this: RealmContext) {
+          const realm = this.realm;
+          // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+          const callback = spy((estimate: number) => {});
+          const realm2 = await Realm.open({
+            schema: [Person, Dog],
+            sync: {
+              flexible: true,
+              user: await this.app.logIn(Credentials.anonymous(false)),
+              newRealmFileBehavior: {
+                type: OpenRealmBehaviorType.OpenImmediately,
+              },
+            },
+          });
+
+          realm2.syncSession?.addProgressNotification(
+            Realm.ProgressDirection.Download,
+            Realm.ProgressMode.ReportIndefinitely,
+            callback,
+          );
+
+          const subscription = await realm.objects(Person).subscribe();
+
+          realm.write(() => {
+            realm.create(Person, {
+              _id: new BSON.ObjectId(),
+              name: "Heavy",
+              age: 36,
+              data: new ArrayBuffer(1_000_000),
+            });
+          });
+
+          const subscription2 = await realm2.objects(Person).subscribe();
+          await realm.syncSession?.uploadAllLocalChanges();
+          await realm2.syncSession?.downloadAllServerChanges();
+
+          expect(callback.args.find(([estimate]) => estimate < 1)).to.not.be.undefined;
+          expect(callback.withArgs(1.0).callCount).is.greaterThanOrEqual(2);
+          const oldCallCount = callback.callCount;
+
+          realm2.syncSession?.removeProgressNotification(callback);
+
+          realm.write(() => {
+            realm.create(Person, {
+              _id: new BSON.ObjectId(),
+              name: "Heavy",
+              age: 36,
+              data: new ArrayBuffer(1_000_000),
+            });
+          });
+
+          await realm.syncSession?.uploadAllLocalChanges();
+          await realm2.syncSession?.downloadAllServerChanges();
+
+          expect(callback).has.callCount(oldCallCount);
```

Comment:
I think earlier the `expect(callback.callCount).to.be.equal(oldCallCount)` was used, we could align there 👍